testsuite/integration-tests/src/test/java/org/jboss/resteasy/test/providers/jsonb/basic/JsonBindingDebugLoggingTest.java

```diff
@@ -0,0 +1,324 @@
+    *
+    * In this scenario the exception originates in JsonBindingProvider.readFrom.
+    * The exception is passed up the call stack to MessageBodyParameterInjector.inject
+    * which identifies it as read exception and wraps it in a ReaderException class.
+    * This exception continues up the call stack to SynchronousDispatcher.writeException.
+    * The ExceptionHandler processes it as a ReaderException.  The exception is
+    * written to the server.log by this class.  A Response object is created.  The
+    * exception message is assigned as the (String) entity in the response object.
+    * The response object is returned to the caller.
+    */
+   @Test
+   public void exceptionDuringServerReceive() throws Exception {
+
+      // count log messages before request
+      LogCounter errorStringLog = new LogCounter("ERROR", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      LogCounter resteasyExceptionLog = new LogCounter(".*DEBUG .* RESTEASY002305.*", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER, true);
+
+      LogCounter yassonExceptionLog = new LogCounter("Caused by: javax.json.bind.JsonbException", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter yassonStacktraceLog = new LogCounter("at org.eclipse.yasson", true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+      LogCounter applicationExcpetionLog = new LogCounter(
+              "Caused by: java.lang.RuntimeException: "
+            + JsonBindingDebugLoggingItemCorruptedSet.class.getSimpleName(), true, ContainerConstants.DEFAULT_CONTAINER_QUALIFIER);
+
+      // perform request
+      WebTarget base = client.target(generateURL("/post"));
+      JsonBindingDebugLoggingItem wrongItem = new JsonBindingDebugLoggingItem();
+      wrongItem.setA(5);
+      Response response = base.request().post(Entity.entity(wrongItem,
+              MediaType.APPLICATION_JSON));
+
+      // check response
+      Assert.assertThat("Wrong response code", response.getStatus(), is(400));
+      Assert.assertThat("Response message doesn't contains expected message",
+              response.readEntity(String.class),
+              containsString("RESTEASY008200: JSON Binding deserialization error"));
+
+      // assert log messages after request in the server.log
+      Assert.assertThat("Application Exception should be logged",
+              applicationExcpetionLog.count(), is(1));
+      Assert.assertThat("RESTEasy exception should be logged",
+              resteasyExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception should be logged",
+              yassonExceptionLog.count(), is(1));
+      Assert.assertThat("Yasson exception stacktrace should be logged",
+              yassonStacktraceLog.count(), greaterThan(0));
+
+      Assert.assertThat("There shouldn't be any error logs in server",
+              errorStringLog.count(), is(0));
+   }
+
+   /**
+    * @tpTestDetails Check exception during server receiving

```

Comment:
"client receiving"

Reply:
Looks like a typo ...

Reply:
@rsearls, shouldn't this be "client receiving"?